docs/create-a-new-angular-component.md

```diff
@@ -0,0 +1,497 @@
+# Create a new Angular App Component
+
+## Overview
+
+This doc describes how to create a new library + component in the Challenge Registry app, though
+the steps can be applied to any app in this project.  This doc will also include information on
+where/how to copy-paste code from the [Figma-to-code export] into the app (starting at [Step 5]).
+
+## 1. Create a new Angular library
+
+To create a UI library within `challenge-registry`, run:
+
+```console
+nx g @nrwl/angular:lib <new library name> --directory challenge-registry
+```
+ 
+(Optional but recommended) Use `--dryrun` to first see what and where the entities will be created;
+this will help visualize and validate the intended directory structure, e.g.
+
+```console
+$ nx g @nrwl/angular:lib awesome-lib --directory challenge-registry --dry-run
+
+>  NX  Generating @nrwl/angular:library
+
+...
+UPDATE workspace.json
+CREATE libs/challenge-registry/awesome-lib/README.md
+CREATE libs/challenge-registry/awesome-lib/tsconfig.lib.json
+CREATE libs/challenge-registry/awesome-lib/tsconfig.spec.json
+CREATE libs/challenge-registry/awesome-lib/src/index.ts
+CREATE libs/challenge-registry/awesome-lib/src/lib/challenge-registry-awesome-lib.module.ts
+CREATE libs/challenge-registry/awesome-lib/tsconfig.json
+CREATE libs/challenge-registry/awesome-lib/project.json
+UPDATE tsconfig.base.json
+CREATE libs/challenge-registry/awesome-lib/jest.config.ts
+CREATE libs/challenge-registry/awesome-lib/src/test-setup.ts
+CREATE libs/challenge-registry/awesome-lib/.eslintrc.json
+
+NOTE: The "dryRun" flag means no changes were made.
+```
+
+Due to how the Challenge Registry app is currently structured, some additional
+steps are required:
+
+1. Discard/undo changes made to `project.json` in other existing folders outside of the new library
+folder, e.g. `apps/challenge-api-gateway/project.json`. The only files you need staged are:
+
+    * files created for the new library, e.g. `libs/challenge-registry/<new library name/*`
+    * `tsconfig.base.json`
+    * `workspace.json`
+
+    You could do something like this:
+    
+    ```
+    git add libs/challenge-registry/awesome-lib/.* tsconfig.base.json workspace.json
```

Comment:
with above snippet, somehow it will also add other packages - my bad. Could you change to the below snippet (which will add the expected files based on my testing)?

```
git add libs/challenge-registry/awesome-lib/ tsconfig.base.json workspace.json
```